app/workers/billing_worker.rb

```diff
@@ -38,14 +38,33 @@ def on_complete(status, options)
     end
   end
 
+  SPARSING_RATE = (2.0/3).freeze # in seconds/transaction
+
+  def self.sparsing_rate
+    SPARSING_RATE
+  end
+
   # @param [Account] provider
   # @param [Time] billing_date
   # @param [ActiveRecord::Relation] buyers_scope
   def self.enqueue(provider, billing_date, buyers_scope = nil)
-    needs_lock = PaymentGateway.find(provider.payment_gateway_type)&.need_lock?
+    payment_gateway = PaymentGateway.find(provider.payment_gateway_type)
+    needs_lock = payment_gateway&.need_lock?
```

Comment:
I think the `Null Object Pattern` would be better for this case. Since the default case is doing nothing, we can do something like this:
```ruby
def self.enqueue(provider, billing_date, buyers_scope = nil)
    payment_gateway = PaymentGateway.find(provider.payment_gateway_type) || PaymentGateway.new(provider.payment_gateway_type)

    with_billing_batch(provider.id, billing_date) do
      count = 0
      provider.buyer_accounts.select(:id, :provider_account_id).merge(buyers_scope).find_in_batches do |group|
        group.each do |buyer|
          options = { needs_lock: payment_gateway.needs_lock? }

          if payment_gateway.need_sparsing?
            options[:delay] = (sparsing_rate * count).seconds
            count += 1
          end

          enqueue_for_buyer(buyer, billing_date, **options)
        end
      end
    end
end
```

Or maybe even better having the build of `payment_gateway` as `payment_gateway = PaymentGateway.find_or_build(provider.payment_gateway_type)`
And that method actually doing inside:
```ruby
def find_or_build(type)
  find(type) || new(type)
end
```